pkg/reconciler/pipelinerun/resources/resultrefresolution.go

```diff
@@ -151,12 +151,12 @@ func resolveResultRef(pipelineState PipelineRunState, resultRef *v1beta1.ResultR
 	}
 
 	var runName, runValue, taskRunName string
-	var resultValue v1beta1.ArrayOrString
+	var resultValue v1beta1.ParamValue
```

Comment:
should this be a ResultValue? I think you may be able to cast the return value (not sure). As an aside when we update to go 1.18 we might be able to use generics for that function.

Reply:
Oh yes! This should be result value. I have went through the `ParamValue` used in result files again and fix some other places as well

Now they can be used interchangeably due to the type alias.